mockito-core/src/test/java/org/mockito/internal/stubbing/InvocationContainerImplTest.java

```diff
@@ -114,4 +114,34 @@ public void should_tell_if_has_invocation_for_potential_stubbing_stub_only() thr
         containerStubOnly.addAnswer(new ReturnsEmptyValues(), null);
         assertFalse(containerStubOnly.hasInvocationForPotentialStubbing());
     }
+
+    @Test
+    public void should_return_answer_when_answer_exists() throws Throwable {
+        // Given: set an invocation for stubbing and add a matching answer
+        container.setInvocationForPotentialStubbing(new InvocationMatcher(invocation));
+        container.addAnswer(new Returns("Expected Answer"), null);
+
+        // When: invoking answerTo should return the answer
+        Object result = container.answerTo(invocation);
+
+        // Then: verify the expected answer is returned
+        assertEquals("Expected Answer", result);
+    }
+
+    @Test
+    public void should_throw_null_pointer_exception_when_no_answer_found() {
+        // Given: no answer added for the invocation
+        Invocation unmatchedInvocation = new InvocationBuilder().toInvocation();
+
+        // When and Then: invoking answerTo should throw NullPointerException
+        NullPointerException exception =
+                assertThrows(
+                        NullPointerException.class,
```

Comment:
This should expect a `MockitoException` instead

Reply:
Ok changes made, im a first time contributor thanks for bearing with me